src/components/Layout.js

```diff
@@ -45,6 +45,29 @@ const startPath = (langKey, langsMenu, basename, _url) => {
   return basePath;
 };
 
+const blog_basename = (langKey, _url) => {
```

Comment:
i created this two functions blog_basename and check_path to solve the issue with the blog ids, Maybe there are other more elegant and efficient solutions, but for now it works...